utils/build_ds_container.py

```diff
@@ -0,0 +1,256 @@
+
+from kubernetes import client, config
+from kubernetes.client import exceptions
+
+config.load_kube_config()
+
+
+DESCRIPTION = '''
+A tool for building compliance content for Kubernetes clusters.
+
+This tool supports building ComplianceAsCode content locally or remotely on an
+existing kubernetes cluster.
+
+'''
+
+HTTP_CONFLICT_STATUS_CODE = 409
+
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    description=DESCRIPTION)
+parser.add_argument(
+    '-n', '--namespace',
+    help='Build image in the given namespace.',
+    default='openshift-compliance')
+parser.add_argument(
+    '-p', '--create-profile-bundles',
+    help='Create ProfileBundle objects for the image.',
+    action='store_true',
+    default=False)
+parser.add_argument(
+    '-c', '--build-in-cluster',
+    help=(
+        'Build content in-cluster. Note that this option ignores the '
+        '--product and --debug flags.'),
+    action='store_true',
+    default=False)
+parser.add_argument(
+    '-d', '--debug',
+    help=(
+        'Provide debug output during the build process. This option is '
+        'ignored when building content in the cluster using '
+        '--build-in-cluster.'),
+    action='store_true',
+    default=False)
+parser.add_argument(
+    '-P', '--product',
+    help=(
+        'The product(s) to build. This option can be provided multiple times. '
+        'This option is ignored when building content in the cluster using '
+        '--build-in-cluster.'),
+    default=['ocp4', 'rhcos4'],
+    dest='products',
+    nargs='*')
+parser.add_argument(
+    '-r', '--repository',
+    help=(
+        'The container image repository to use for images containing built '
+        'content. Images pushed to this repository must have a tag, which '
+        'you can specify with the --container-image-tag argument. If you do '
+        'not supply a tag, one will be generated for you. It is recommended '
+        'that you properly tag built images for production use. '
+        'Auto-generated tags are primarily useful for development workflows. '
+        'This script assumes you have authenticated to the image repository '
+        'if necessary (e.g, `podman login`).'))
+parser.add_argument(
+    '-t', '--container-image-tag',
+    help='A unique tag for the container image.')
+args = parser.parse_args()
+
+
+log.basicConfig(
+    format='%(asctime)s:%(levelname)s: %(message)s', level=log.INFO)
+
+
+# FIXME(lbragstad): Remove this and replace it with operating system utils if
+# possible.
+result = subprocess.run(
+    ['git', 'rev-parse', '--show-toplevel'], capture_output=True, check=True)
+# Convert the file path from bytes to unicode since we might manipulate it
+# later. Also, strip off any newlines.
+REPO_PATH = result.stdout.decode().strip()
+
+
+def ensure_namespace_exists(namespace):
+    """Function that ensures there is a namespace for the content."""
+    with client.ApiClient() as api_client:
```

Comment:
I have to admit, I'm not a big fan of using full blown kube APIs for simple scripts. using `subprocess.run` alongside `kubectl` or `oc` would be both more simple, and would make the declarations more obvious (since it would rely on YAML), thus easier to maintain. You could then do `oc apply -f ...` would would already take care of exception of objects already existing and such. So all of this would be reduced to just a couple of lines.